Common/cpp/SharedItems/ShareableValue.cpp

```diff
@@ -213,22 +230,24 @@ jsi::Value ShareableValue::toJSValue(jsi::Runtime &rt) {
         // call on an appropriate thread
         
         auto module = this->module;
-        auto hostFunction = this->hostFunction;
+        std::weak_ptr<HostFunctionHandler> hostFunction = this->hostFunction;
```

Comment:
It cannot be a weak_ptr because the function will be instantly deallocated. Imagine the following example: 
```JS
runOnJS(worklet)(callback_hostfunction); // 
```

Reply:
using worklet in ruOnJS results in an error

Reply:
You probably meant something like this:
```
const wrk = (clb) => {
  'worklet';
  runOnJS(clb)();
};

const hf = () => {}

runOnUI(wrk)(hf);
```

Reply:
I meant runOnUI.

Reply:
discussed internally